gmapping/src/slam_gmapping.cpp

```diff
@@ -628,7 +633,9 @@ SlamGMapping::laserCallback(const std::shared_ptr<sensor_msgs::msg::LaserScan> s
     map_to_odom_ = (odom_to_laser * laser_to_map).inverse();
     map_to_odom_mutex_.unlock();
 
-    if(!got_map_ || (tf2_ros::fromMsg(rclcpp::Time::now()) - last_map_update) > map_update_interval_)
+    tf2::TimePoint stamp_time = tf2_ros::fromMsg(scan->header.stamp);
+
+    if(!got_map_ || (stamp_time - last_map_update) > map_update_interval_)
```

Comment:
This line, and line 603 were the only really significant differences I noticed between the branches.

Your branch was comparing the current time to the last map update time, original compares the scan timestamp (instead of now).

Original also initialized last_map_update to zero time, which probably doesn't effect much.